generator/src/test/java/com/linkedin/pegasus/generator/TestProjectionMaskApiChecker.java

```diff
@@ -0,0 +1,165 @@
+package com.linkedin.pegasus.generator;
+
+import com.linkedin.data.DataMap;
+import com.linkedin.data.schema.DataSchemaLocation;
+import com.linkedin.data.schema.MaskMap;
+import com.linkedin.data.schema.RecordDataSchema;
+import com.linkedin.data.template.RecordTemplate;
+import com.linkedin.pegasus.generator.spec.ClassTemplateSpec;
+import com.sun.codemodel.JClass;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+
+public class TestProjectionMaskApiChecker {
```

Comment:
This unit test looks good to me. 

BTW, how did you find the rollout problem? Have you E2E tests in a depender of Pegasus? 

Or maybe we should do that later

Reply:
I didn't run E2E tests yet. Found it when I was thinking about rollout.